Source/KSCrash/Reporting/Tools/KSHTTPMultipartPostBody.m

```diff
@@ -161,28 +163,34 @@ - (NSData*) data
     }
 
     NSMutableData* data = [NSMutableData dataWithCapacity:baseSize];
+    BOOL firstFieldSent = NO;
     for(KSHTTPPostField* field in _fields)
     {
-        [data appendUTF8Format:@"--%@\r\n", g_boundary];
+        if (firstFieldSent) {
+            [data appendUTF8Format:@"\r\n"];
+        } else {
+            firstFieldSent = YES;
+        }
+        [data appendUTF8Format:@"--%@\r\n", _boundary];
         if(field.filename != nil)
         {
             [data appendUTF8Format:@"Content-Disposition: form-data; name=\"%@\"; filename=\"%@\"\r\n",
-             [field.name URLEncoded],
-             [field.filename URLEncoded]];
+             field.name,
+             field.filename];
```

Comment:
Well, the [HTML RFC](http://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2) specifies that the value should contain quoted string, meaning that anything is allowed and the quotation mark must be preceded with a backslash.
URL encoding is definitely not correct here as it breaks names like "upload[]".
I believe it's upon the caller to pass the correct value for the fields, anyway.